projects/plugins/crm/includes/ZeroBSCRM.FormatHelpers.php

```diff
@@ -1170,105 +1170,132 @@ function zeroBSCRM_html_taskDate($task=array()){
+			margin-bottom:0.8em;
+		}
+		.zbs-email-sending-record .avatar{
+			margin-left: 5px;
+			border-radius: 50%;
+		}
+		.zbs-email-detail {
+			width: 80%;
+			display: inline-block;
+		}
 	</style>
-    <?php
+	<?php
+
+	if ( count( $email_hist ) === 0 ) {
+		echo '<div class="ui message"><i class="icon envelope outline"></i>' . esc_html( __( 'No Recent Emails', 'zero-bs-crm' ) ) . '</div>';
+	}
+
+	foreach ( $email_hist as $em_hist ) {
+
+		$email_details_html = '';
+
+		$email_subject = zeroBSCRM_mailTemplate_getSubject( $em_hist->zbsmail_type );
+		$emoji         = '🤖';
+
+		if ( $email_subject === '' ) {
+			// then this is a custom email
+			$email_subject = $em_hist->zbsmail_subject;
+			$emoji         = '😀';
+		}
+
+		// if still empty
+		if ( empty( $email_subject ) ) {
+			$email_subject = esc_html__( 'Untitled', 'zero-bs-crm' );
+		}
+		$email_details_html .= '<div class="zbs-email-sending-record">';
+		$email_details_html .= '<span class="label blue ui tiny hist-label" style="float:left"> ' . esc_html( __( 'sent', 'zero-bs-crm' ) ) . ' </span>';
+		$email_details_html .= '<div class="zbs-email-detail">' . esc_html( $emoji );
+		$email_details_html .= ' <strong>' . esc_html( $email_subject ) . '</strong><br />';
+		$email_details_html .= '<span class="sent-to">' . esc_html( __( ' sent to ', 'zero-bs-crm' ) ) . '</span>';
 
-    if(count($email_hist) == 0){
-    	echo "<div class='ui message'><i class='icon envelope outline'></i>" . esc_html( __('No Recent Emails','zero-bs-crm') ) . "</div>";
-    }
-    
-    foreach($email_hist as $em_hist){
-
-        $email_subject = zeroBSCRM_mailTemplate_getSubject($em_hist->zbsmail_type);
-        $emoji = '🤖';
-        if($email_subject ==''){
-          //then this is a custom email
-          $email_subject = $em_hist->zbsmail_subject;
-          $emoji ='😀';
-        }
-        // if still empty
-        if (empty($email_subject)) $email_subject = __('Untitled','zero-bs-crm');
-        echo "<div class='zbs-email-sending-record'>";
-		echo "<span class='label blue ui tiny hist-label' style='float:left'> " . esc_html( __('sent','zero-bs-crm') ) . ' </span>';
-		echo '<div class="zbs-email-detail">'. esc_html( $emoji );
-		echo " <strong>" . esc_html( $email_subject ) . "</strong><br />";
-		echo "<span class='sent-to'>" . esc_html( __(" sent to ", 'zero-bs-crm') ) . "</span>";
 		// -10 are the system emails sent to CUSTOMERS
-		if($em_hist->zbsmail_sender_wpid == -10){
-			$customer = zeroBS_getCustomerMeta($em_hist->zbsmail_target_objid);
-			$link = admin_url('admin.php?page='.$zbs->slugs['addedit'].'&action=view&zbsid=' .$em_hist->zbsmail_target_objid);
-			if($customer['fname'] == '' && $customer['lname'] == ''){
-				echo "<a href='".esc_url( $link )."'>" . esc_html( $customer['email'] ) . "</a>";
-			}else{ 
-				echo "<a href='".esc_url( $link )."'>" . esc_html( $customer['fname'] . ' ' . $customer['lname'] ) . "</a>";
+		if ( $em_hist->zbsmail_sender_wpid === '-10' ) {
```

Comment:
Magic numbers! (I know it's legacy)

Reply:
Worse yet, magic strings (they're numbers in a string format)!

Reply:
😱